documentation/guide-csrf.asciidoc

```diff
@@ -0,0 +1,55 @@
+:toc: macro
+toc::[]
+
+= Cross-site request forgery (CSRF)
+
+CSRF is a type of malicious exploit of a web application that allows an attacker to induce users to perform actions that they do not intend to perform.
+
+image::images/csrf.png[,width="450", link="images/jwt_flow.png"]
+
+More details about csrf
+[source,URL]
+https://owasp.org/www-community/attacks/csrf
+
+Since devon4j uses 'form based authentication', CSRF or JWT  should be enabled.
+How to enable JWT is explained in JWT guide.
+
+=== Dependency
+
+To use CSRF support from devon4j, add following required dependency.
+
+
+[source,xml]
+----
+<dependency>
+      <groupId>com.devonfw.java.starters</groupId>
+      <artifactId>devon4j-starter-security-csrf</artifactId>
+</dependency>
+----
+
+=== Configure CSRF
+
+To enable CSRF protection, make following changes in the BaseWebSecurityConfig:
+
+[source,java]
+----
+ public void configure(HttpSecurity http) throws Exception {
+ .....
+ http
+ .....
+	.csrf().requireCsrfProtectionMatcher(new CsrfRequestMatcher())
+ ....
+ if (this.corsEnabled) {
+    http.addFilterBefore(getCorsFilter(), CsrfFilter.class);
+  }
+ }
+----
+
+Also remove  .csrf.disable()  if exists.
+
+Configure environment.ts file to use the CSRF in the front end.
+
+[source]
+----
+security: 'csrf'
+----
```

Comment:
IMHO we should better link directly to the `devon4ng` guide explaining this option. If that setting changes we will miss it in `devon4j` and the documentation is "broken".